### PR TITLE
Scene as Injectable Service

### PR DESCRIPTION
**Pull request by jaimeatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
The Scene static class has been changed so now its an injectable service in the CanvasComponent

### CODE REVIEW

File: COMETwebapp.Tests/Viewer/PrimitivesTests.cs [hunk 1]
```diff
@@ -76,6 +76,7 @@ public void SetUp()
             var session = new Mock<ISessionAnchor>();
             this.context.Services.AddSingleton(session.Object);
 
+            this.context.Services.AddTransient<ISceneProvider, Scene>();
```
Comment (samatrhea) [inline COMETwebapp.Tests/Viewer/PrimitivesTests.cs:79]: it is good practice to name the implemenation of an interface in the same way, I propose you rename `Scene` to `SceneProvider`

File: COMETwebapp/Components/Viewer/BabylonCanvasBase.cs [hunk 2]
```diff
@@ -70,6 +65,12 @@ public class BabylonCanvasBase : ComponentBase
         [Inject]
         public IShapeFactory? ShapeFactory { get; set; }
 
+        /// <summary>
+        /// The babylon.js scene
+        /// </summary>
+        [Inject]
+        public ISceneProvider Scene { get; set; }
```
Comment (samatrhea) [inline COMETwebapp/Components/Viewer/BabylonCanvasBase.cs:72]: i would rename the parameter from `Scene` to `SceneProvider`

File: COMETwebapp/Components/Viewer/Scene.cs [hunk 3]
```diff
@@ -32,16 +32,18 @@ namespace COMETwebapp
...
     using Microsoft.AspNetCore.Components;
+    using Microsoft.JSInterop;
+
+    using Newtonsoft.Json;
 
     /// <summary>
     /// Static class to access the resources of a Scene 
     /// </summary>
-    public static class Scene
+    public class Scene : ISceneProvider
```
Comment (samatrhea) [inline COMETwebapp/Components/Viewer/Scene.cs:46]: please rename this class to `SceneProvider`

**Review by samatrhea — APPROVED** [review]

**Comment by jaimeatrhea** [comment]
TODO: Use the current GUID structure in C# and the converted string in JS

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**